data_model/src/asset.rs

```diff
@@ -150,21 +150,31 @@ pub enum AssetValueType {
     Store,
 }
 
-impl FromStr for AssetValueType {
-    type Err = &'static str;
-
-    fn from_str(value_type: &str) -> Result<Self, Self::Err> {
-        // TODO: Could be implemented with some macro
-        match value_type {
-            "Quantity" => Ok(AssetValueType::Quantity),
-            "BigQuantity" => Ok(AssetValueType::BigQuantity),
-            "Fixed" => Ok(AssetValueType::Fixed),
-            "Store" => Ok(AssetValueType::Store),
-            _ => Err("Unknown variant"),
+/// A declarative macro that implements `FromStr` for a given
+/// C like enumeration. The macro is invoked like follows:
+/// `easy_from_str_impl! { NameOfEnum, EnumVariation1, EnumVariation2, ... }`
+macro_rules! easy_from_str_impl {
+    (eval_to $cmp:expr, $enum_type:ty, $enum_value:tt) => {
+        if $cmp == stringify!($enum_value) {
+            return Ok(<$enum_type>::$enum_value);
         }
-    }
+    };
+    ($enum_type:ty, $( $enum_value:tt ),+ ) => {
+        impl FromStr for $enum_type {
+            type Err = &'static str;
+
+            fn from_str(value_type: &str) -> Result<Self, Self::Err> {
+                $(
+                    easy_from_str_impl!{eval_to value_type, $enum_type, $enum_value}
+                )+
+                return Err(concat!("Unknown variant for type ", stringify!($enum_type)));
+            }
+        }
+    };
 }
 
+easy_from_str_impl! {AssetValueType, Quantity, BigQuantity, Fixed, Store}
```

Comment:
Just to mention for future macros. I think that's not a good tradition in Iroha to write such simple to implement but hard to read syntax.

Take a look at this [rule](https://rust-lang.github.io/api-guidelines/macros.html):

> Rust macros let you dream up practically whatever input syntax you want. Aim to keep input syntax familiar and cohesive with the rest of your users' code by mirroring existing Rust syntax where possible. Pay attention to the choice and placement of keywords and punctuation.